### PR TITLE
DEV: Convert to plugin

### PR DESCRIPTION
**Pull request by jbrw**

### CODE REVIEW

File: plugin.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# name: DiscourseOnboardingBanner
+# about: Display a dismissable banner during onboarding
+# version: 0.1
+# authors: awesomerobot and Jamie Wilson
+# url:https://github.com/discourse/discourse-onboarding-banner
+
```
Comment (CvX) [inline plugin.rb:8]:
```suggestion
# transpile_js: true

```

File: plugin.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# name: DiscourseOnboardingBanner
+# about: Display a dismissable banner during onboarding
+# version: 0.1
+# authors: awesomerobot and Jamie Wilson
+# url:https://github.com/discourse/discourse-onboarding-banner
```
Comment (CvX) [inline plugin.rb:7]:
```suggestion
# url: https://github.com/discourse/discourse-onboarding-banner
```

File: plugin.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+register_asset 'stylesheets/common/discourse-onboarding-banner.scss'
+
+enabled_site_setting :discourse_onboarding_banner_enabled
+
+PLUGIN_NAME ||= 'DiscourseOnboardingBanner'
```
Comment (CvX) [inline plugin.rb:13]:
```suggestion
PLUGIN_NAME = 'DiscourseOnboardingBanner'
```

File: assets/javascripts/discourse/templates/components/onboarding-banner.hbs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{{#unless maxExpired}}
+<div class="onboarding-banner-content">
+  {{html-safe cooked}}
+  {{d-button icon="times" action=(action "dismissOnboarding")}}
+</div>
```
Comment (CvX) [inline assets/javascripts/discourse/templates/components/onboarding-banner.hbs:5]:
```suggestion
  <div class="onboarding-banner-content">
    {{html-safe cooked}}
    {{d-button icon="times" action=(action "dismissOnboarding")}}
  </div>
```

File: assets/javascripts/discourse/components/onboarding-banner.js.es6 [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import { inject as service } from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import PostCooked from "discourse/widgets/post-cooked";
+
+export default Component.extend({
+  classNameBindings: ["onboarding-banner"],
+  router: service(),
+  cooked: null,
+  maxExpired: false,
+
```
Comment (CvX) [inline assets/javascripts/discourse/components/onboarding-banner.js.es6:11]:
```suggestion
  isLoading: true,

```

File: assets/javascripts/discourse/components/onboarding-banner.js.es6 [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    } else {
+      if (storageObject) {
+        this.set("cooked", storageObject.onboarding_topic);
+        this.set("isLoading", false);
+      }
+    }
+  },
+
+  actions: {
+    dismissOnboarding() {
```
Comment (CvX) [inline assets/javascripts/discourse/components/onboarding-banner.js.es6:101]: Please use `@action` decorator 😃

File: test/javascripts/acceptance/discourse-onboarding-banner-test.js.es6 [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+
+acceptance("DiscourseOnboardingBanner", { loggedIn: true });
+
+test("DiscourseOnboardingBanner works", async assert => {
+  await visit("/admin/plugins/discourse-onboarding-banner");
+
+  assert.ok(false, "it shows the DiscourseOnboardingBanner button");
+});
```
Comment (CvX) [inline test/javascripts/acceptance/discourse-onboarding-banner-test.js.es6:9]:
```suggestion
import { acceptance } from "discourse/tests/helpers/qunit-helpers";
import { test } from "qunit";
import { visit } from "@ember/test-helpers";

acceptance("DiscourseOnboardingBanner", function (needs) {
  needs.user();

  test("DiscourseOnboardingBanner works", async assert => {
    await visit("/admin/plugins/discourse-onboarding-banner");

    assert.ok(false, "it shows the DiscourseOnboardingBanner button");
});
```

And the assertion needs to be replaced with the real thing 😃

**Review by CvX — CHANGES_REQUESTED** [review]
I might have jump the gun on the review 😅 Anyway, a todo list won't hurt 😛

In addition to inline suggestions, can you update package.json and .gitignore, and add .github/workflows, .prettierc, .rubocop.yml, Gemfile, Gemfile.lock (see https://github.com/discourse/discourse-plugin-skeleton/), and rename *.es6 files to *.js? 😃

**Review by CvX — COMMENTED** [review]
Just couple more things 😃

File: test/javascripts/acceptance/discourse-onboarding-banner-test.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import { visit } from "@ember/test-helpers";
```
[the remark this answers is not in the document]
Comment (CvX) [reply test/javascripts/acceptance/discourse-onboarding-banner-test.js:3]: Oh, and `exists` (and visible/invisible) needs to be imported from qunit-helpers

File: test/javascripts/acceptance/discourse-onboarding-banner-test.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  test("DiscourseOnboardingBanner is shown and can be dismissed", async (assert) => {
+    await visit("/");
+    assert.ok(
+      exists(".onboarding-banner-content"),
+      "it shows the DiscourseOnboardingBanner"
+    );
+
+    await click(".onboarding-banner-content .dismiss-banner");
+    assert.ok(
+      !exists(".onboarding-banner-content:visible"),
```
[the remark this answers is not in the document]
Comment (jbrw) [reply test/javascripts/acceptance/discourse-onboarding-banner-test.js:34]: Neat! I wasn't aware of `invisible()`/`visible()`.

File: assets/javascripts/discourse/components/onboarding-banner.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+              firstSeen: storageObject ? storageObject.firstSeen : date,
+            };
+
+            localStorage.setItem(
+              "onboarding_topic",
+              JSON.stringify(dataObject)
+            );
+
+            this.set("cooked", cachedTopic.attrs.cooked);
+        })
```
Comment (CvX) [inline assets/javascripts/discourse/components/onboarding-banner.js:88]:
```suggestion
            this.set("cooked", cachedTopic.attrs.cooked);
        })
      }
```

Oops, I broke the code :P

File: assets/javascripts/discourse/components/onboarding-banner.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+    ajax("/discourse-onboarding-banner/dismiss.json", {
+      type: "PUT",
+      data,
+    })
+      .finally(() => {
+        document.querySelector("div.onboarding-banner").style.display = "none";
+        localStorage.removeItem("onboarding_topic");
+      })
+      .catch(() => {});
```
[the remark this answers is not in the document]
Comment (jbrw) [reply assets/javascripts/discourse/components/onboarding-banner.js]: That was my bad - the error that was being thrown was due to a bad stubbed response. Fixing that cleared up the error that was causing the issues.

**Review by CvX — APPROVED** [review]

**Comment by jbrw** [comment]
Thank you @CvX for helping get this cleaned up and ready to ship!

**Pull request merged**